File: src/py3/043-c/043-c.py
#!/usr/bin/env python3
# -*- coding: utf-8 -*-

"""043-c"""


import sys

from functools import reduce


def main():
    """Main function."""
    N = int(input())
    a_list = list(map(int, input().split()))

    # print(N, a_list)

    x = sum(a_list) / N

    def sumOfProd(n,l):
        """Sum of product."""
        return reduce(lambda a,b: a + b,
                      map(lambda v: (v - n) * (v - n), l))

    res1 = sumOfProd(int(x), a_list)
    res2 = sumOfProd(int(x) + 1, a_list)

    print(min(res1, res2))


if __name__ == '__main__':
    sys.exit(main())

File: src/cpp14/043-c/043-c.cpp
#include <iostream>
#include <cmath>
#include <vector>

using namespace std;

int main(int argc, char const* argv[])
{
  int N;
  cin >> N;
  vector<int> a_list(N);

  int sum = 0;
  for (size_t i = 0; i < N; i++) {
    cin >> a_list[i];
    sum += a_list[i];
  }

  int x = sum / N;

  int res1 = 0;
  int res2 = 0;

  for (size_t i = 0; i < N; i++) {
    res1 += (x - a_list[i]) * (x - a_list[i]);
    res2 += (x + 1 - a_list[i]) * (x + 1 - a_list[i]);
  }

  if (res1 <= res2) {
    cout << res1;
  } else {
    cout << res2;
  }
  cout << endl;

  return 0;
}
